script.js

```diff
@@ -0,0 +1,218 @@
+                        data-toggle="dropdown" aria-haspopup="true" aria-expanded="false" >
+                    <i class="fas fa-ellipsis-v"></i>
+                </button>
+                <div class="dropdown-menu p-2 flex-column" aria-labelledby="dropdownMenuItem1" id="${task.id}">
+                    ${!task.isDone ? `<button type="button" class="btn btn-success w-100">Complete</button>
+    <button type="button" class="btn btn-info w-100 my-2" data-toggle="modal" data-target="#exampleModal">Edit</button>`
+        : `<button type="button" class="btn btn-success w-100 mb-2">Uncomplete</button>`}
+                    <button type="button" class="btn btn-danger w-100">Delete</button>
+                </div>
+            </div>`;
+    if (sortUp) {
+        task.isDone ? completedTasksBlock.append(taskLayout) : currentTasksBlock.append(taskLayout);
+    } else {
+        task.isDone ? completedTasksBlock.prepend(taskLayout) : currentTasksBlock.prepend(taskLayout);
+    }
+}
+
+function renderApp() {
+    currentTasksBlock.innerHTML = null;
+    completedTasksBlock.innerHTML = null;
+    allTasks.map(task => addTaskLayout(task));
+    const tasksCount = getTasksCount();
+    currentTasksTitle.textContent = `ToDo (${tasksCount.current})`;
+    completedTasksTitle.textContent = `Completed (${tasksCount.completed})`;
+}
+
+function removeTask(taskId) {
+    const copyTasks = allTasks.filter(task => task.id !== taskId);
+    updateLocalStorage(LocalStorageKeys.tasks, copyTasks);
+    allTasks = copyTasks;
+    renderApp();
+}
+
+function changeTaskStatus(taskId) {
+    const copyTasks = allTasks.map(task => task.id === taskId ? {...task, isDone: !task.isDone} : task);
+    updateLocalStorage(LocalStorageKeys.tasks, copyTasks);
+    allTasks = copyTasks;
+    renderApp();
+}
+
+function showEditModal(taskId) {
+    const task = allTasks.find(task => task.id === taskId);
+    modalLabelText.textContent = TEXT_VARIABLES.editTask;
+    submitButton.textContent = TEXT_VARIABLES.save;
+    inputTitle.value = task.title;
+    inputText.value = task.text;
+    form.elements[task.priority].checked = true;
+    editTaskId = taskId;
+    modalEditMode = true;
+}
+
+
+function tasksBlockListener({target}) {
+    const dropDownMenu = target.closest('.dropdown-menu');
+    if (dropDownMenu) {
+        const taskId = dropDownMenu.getAttribute('id');
+        if (target.classList.contains(BUTTON_CLASSNAMES.editBtn)) {
+            showEditModal(taskId);
+        } else if (target.classList.contains(BUTTON_CLASSNAMES.changeStatusBtn)) {
+            changeTaskStatus(taskId);
+        } else {
+            removeTask(taskId);
+        }
+    }
+}
+
+currentTasksBlock.addEventListener('click', tasksBlockListener);
+completedTasksBlock.addEventListener('click', tasksBlockListener);
+
+function closeModal() {
+    modalLabelText.textContent = TEXT_VARIABLES.addTask;
+    submitButton.textContent = TEXT_VARIABLES.addTask;
```

Comment:
You don't need to change headers to default every time. It should depend on modal type you open.

Reply:
``
```suggestion
addTaskButton.addEventListener('click',()=>{
    modalLabelText.textContent = TEXT_VARIABLES.addTask;
    submitButton.textContent = TEXT_VARIABLES.addTask;
    mediumPriority.checked = true;
    inputTitle.value = '';
    inputText.value = '';
    modalEditMode = false;
})
```
Removed all cleanup events, now form clears when user open it.